cpp/include/cugraph/graph_functions.hpp

```diff
@@ -875,4 +876,23 @@ weight_t compute_total_edge_weight(
   graph_view_t<vertex_t, edge_t, store_transposed, multi_gpu> const& graph_view,
   edge_property_view_t<edge_t, weight_t const*> edge_weight_view);
 
+/**
+ * @brief Select random vertices
+ *
+ * @tparam vertex_t Type of vertex identifiers. Needs to be an integral type.
+ * @param  handle RAFT handle object to encapsulate resources (e.g. CUDA stream, communicator, and
+ * handles to various CUDA libraries) to run graph algorithms.
+ * @param  rng_state The RngState instance holding pseudo-random number generator state.
+ * @param  num_vertices The number of vertices in the graph
+ * @param  select_count The number of vertices to select from the graph
+ * @param  with_replacement If true, select with replacement, if false select without replacement
+ * @return Device vector of selected vertices.
+ */
+template <typename vertex_t>
+rmm::device_uvector<vertex_t> select_random_vertices(raft::handle_t const& handle,
+                                                     raft::random::RngState& rng_state,
+                                                     vertex_t num_vertices,
+                                                     vertex_t select_count,
+                                                     bool with_replacement);
```

Comment:
This is assuming that vertex IDs are consecutive integers starting from 0. This won't work once we start to support vertex masking. To be future proof, should we better take a graph_view object and call `number_of_vertices()` inside? Then, once we start to support vertex masking, we just need to update the function internals, and users don't need to make change in their side.

Reply:
I'll work on this.  I hadn't thought about that particular use case, I had been thinking it might be better to pass in the vertex, but didn't really have a good use case to motivate it.

Reply:
The legacy implementation of betweenness has a similar python based [function](https://github.com/rapidsai/cugraph/blob/branch-23.02/python/cugraph/cugraph/centrality/betweenness_centrality.py#L282) which I am obviously going to substitute with the CAPI one from sampling_algorithms.